lib/tasks/bulk.rake

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+namespace :bulk do
+  desc "Ingest a directory of TIFFs as a ScannedResource, or a directory of directories as a MultiVolumeWork"
+  task ingest: :environment do
+    user = User.find_by_user_key(ENV['USER']) if ENV['USER']
+    user = User.all.select(&:admin?).first unless user
+    dir = ENV['DIR']
+    bib = ENV['BIB']
+    coll = ENV['COLL']
+    local_id = ENV['LOCAL_ID']
+    replaces = ENV['REPLACES']
+    background = ENV['BACKGROUND']
+    abort "usage: rake bulk:ingest DIR=/path/to/files BIB=1234567 COLL=collid LOCAL_ID=local_id REPLACES=replaces" unless dir && Dir.exist?(dir)
+
+    if coll.present?
+      query = FindByTitle.new(query_service: Valkyrie.config.metadata_adapter.query_service)
+      collection = query.find_by_title(title: coll)
+    end
+
+    @logger = Logger.new(STDOUT)
+    @logger.warn "No BIB id specified" unless bib
+    @logger.info "ingesting files from: #{dir}"
+    @logger.info "ingesting as: #{user.user_key} (override with USER=foo)"
+    @logger.info "adding item to collection #{coll}" if coll
+
+    begin
+      if background
+        IngestDiskFolderJob.perform_later(directory: dir, collection: coll, source_metadata_identifier: bib, local_identifier: local_id, replaces: replaces)
+      else
+        change_set_persister = PlumChangeSetPersister.new(
```

Comment:
Maybe just do `perform_now` to avoid duplicating the job logic here?

Reply:
I meant that lines 30-37 could just be `IngestDiskFolderJob.perform_now` with the same arguments as the `perform_later` call above.  Line 28 should still be `perform_later` since `background` is true.